### ISSUE TITLE
[Bug]: (ExpandableTile) Warning: validateDOMNesting(...): <button> cannot appear as a descendant of <button>.

### ISSUE THREAD

**Issue by EmelyanovSI** [carbon-design-system/carbon#13361]
### Package

@carbon/react

### Browser

Chrome, Firefox, Edge

### Package version

v11.24.0

### React version

v17.0.1

### There is a warning in console when we using `ExpandableTile` with interactive content in dev environment:

![image](https://user-images.githubusercontent.com/48243046/225612041-a57fc77f-fdf9-4671-b3c8-3e736eec4c1c.png)

### But in general there is no button nesting in dom tree:

![image](https://user-images.githubusercontent.com/48243046/225612436-7011494d-a309-4b45-822f-dd30b8da231e.png)

### Related issues, PRs and discussions

- https://github.com/carbon-design-system/carbon/discussions/9003
- https://github.com/carbon-design-system/carbon/issues/9442
- https://github.com/carbon-design-system/carbon/pull/11020

### Reproduction/example

https://codesandbox.io/p/github/EmelyanovSI/carbon-tile-issue

### Steps to reproduce

1. Navigate reproduction/example [link](https://codesandbox.io/p/github/EmelyanovSI/carbon-tile-issue)
2. Open devtools console (F12)

### Suggested Severity

Severity 4 = Unrelated to a user task, has a workaround or does not need a workaround.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
